hera_cal/vis_clean.py

```diff
@@ -213,6 +214,70 @@ def apply_calibration(self, input_cal, unapply=False):
         apply_cal.calibrate_in_place(self.data, cal_gains, self.flags, cal_flags,
                                      gain_convention=gain_convention)
 
+    def apply_flags(self, external_flags=None, overwrite_data_flags=False,
```

Comment:
I would consider keeping the `apply_` methods consistent with how I have them for `apply_calibration(self, input_cal)`.  Note that there I do not change the data in `self.hd` which is kept in its original form, but only change the DataContainers `self.data` and `self.flags`. Unless you think there is a specific reason why we need to change `self.hd`. For writing data, the `self.write_data` method should handle writing the data and flags in `self.data` and `self.flags`

One easy way to do this is to do something like
```
hd_flagged = copy.deepcopy(self.hd)
uvutils.apply_flags(hd_flagged, uvflags)
d,f,n = hd_flagged.build_datacontainers()
self.flags = self.flags + f
```
or something like this

Reply:
Good idea; I think its good to track the original data state.